crates/sui-framework/sources/hex.move

```diff
@@ -38,4 +66,30 @@ module sui::hex {
         assert!(b"fe" == encode(x"fe"), 0);
         assert!(b"00" == encode(x"00"), 0);
     }
+
+    #[test]
+    fun test_hex_decode_string_literal() {
+        assert!(x"ff" == decode(b"ff"), 0);
+        assert!(x"fe" == decode(b"fe"), 0);
+        assert!(x"00" == decode(b"00"), 0);
+    }
```

Comment:
Can you add some test cases which involve multiple bytes, to demonstrate the endianness is working correctly, i.e. the byte order doesn't get flipped?

Reply:
I think I captured most of your feedback / suggestions, but happy to make further changes if I missed or misunderstood anything.